thunder/tests/distributed/test_tensor_parallel.py

```diff
@@ -0,0 +1,195 @@
+        )
+        if bias:
+            expected_bias_grad: torch.Tensor = ref_model.net2.bias.grad
+            if name == _COL:
+                expected = torch.chunk(expected_bias_grad, self.world_size, 0)[self.rank]
+            else:
+                expected = expected_bias_grad
+            torch.testing.assert_close(
+                expected=expected,
+                actual=tp_jitted_model.get_parameter("net2.bias").grad,
+                msg=f"{expected.shape=}, {tp_jitted_model.get_parameter('net2.bias').grad}",
+            )
+        torch.testing.assert_close(
+            expected=(ref_model.net1.weight.grad,) + (ref_model.net1.bias.grad,) if bias else (),
+            actual=(
+                (tp_jitted_model.get_parameter("net1.weight").grad,)
+                + (tp_jitted_model.get_parameter("net1.bias").grad,)
+                if bias
+                else ()
+            ),
+        )
+
+    @pytest.mark.skipif(torch.cuda.device_count() < 2, reason="")
+    @common_utils.parametrize("name", tuple(_name_to_transform.keys()))
+    def test_tensor_parallel_embedding(self, name):
+        num_embeddings = 128
+        embedding_dim = 32
+
+        class Model(nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.embed = nn.Embedding(num_embeddings, embedding_dim)
+
+            def forward(self, x):
+                return self.embed(x)
+
+        device = torch.device(f"cuda:{self.rank}")
+        x = torch.randint(0, num_embeddings - 1, (16, 16), device=device)
+        x_ref = x.clone().detach()
+
+        process_group = None
+        ref_model = Model().to(device)
+
+        ref_state_dict = ref_model.state_dict()
+        expected = ref_model(x_ref)
+
+        transform = _name_to_transform[name]
+        model = Model().to(device)
+        model.load_state_dict(ref_state_dict)
+        jitted_model = thunder.jit(model)
+        tp_jitted_model = transform(
+            jitted_model,
+            target_modules=("embed",),
+            process_group=process_group,
+        )
+        y = tp_jitted_model(x)
+
+        dim: int
+        orig_size: int
+        if name == _COL:
+            dim = 0
+            orig_size = num_embeddings
+        else:
+            dim = 1
+            orig_size = embedding_dim
+        torch.testing.assert_close(
+            tp_jitted_model.get_parameter("embed.weight").size(dim), orig_size // self.world_size
+        )
+        torch.testing.assert_close(expected=expected, actual=y)
+
+        expected.mean().backward()
+        y.mean().backward()
+
+        torch.testing.assert_close(
+            expected=ref_model.embed.weight.grad.chunk(self.world_size, dim)[self.rank],
+            actual=tp_jitted_model.get_parameter("embed.weight").grad,
+        )
+
+    @pytest.mark.skipif(torch.cuda.device_count() < 2, reason="")
+    @common_utils.parametrize("bias", (True, False))
+    def test_tensor_parallel_both_column_and_row(self, bias):
```

Comment:
one more fix is needed. the grads don't match

Reply:
Great job finding and fixing it!